src/plopp/backends/matplotlib/image.py

```diff
@@ -76,14 +81,18 @@ class Image:
 
     def __init__(
         self,
+        uid: str,
         canvas: Canvas,
+        colormapper: ColorMapper,
         data: sc.DataArray,
         shading: str = 'auto',
         rasterized: bool = True,
         **kwargs,
     ):
         check_ndim(data, ndim=2, origin='Image')
+        self.uid = uid
```

Comment:
Why did you make `uid` a function argument? It seems risky to require the user to provide unique ids.

Reply:
> I has now become useful to have ids so that the colormapper could easily access or remove artists under its management (for example when you remove a set of scatter points).

Would this not be possible with an interface like this? That is, the id is generated by the artist but accessed by the colormapper?
```python
artist = make_artist()
self.artists[artist.uid] = artist
```

> But maybe there is a better way of achieving this? The colormapper could have a list of artists instead of a dict. Maybe we almost never need to remove artists?

Depends on how the artists are usually accessed. If we rarely need to access a specific one by id but usually just loop over all, then a list would be fine.

Reply:
> Would this not be possible with an interface like this?

This would be ok for the colormapper, but it wouldn't work for the `GraphicalView` as for consistency, we need the artist `uid` to be the same as the `key` passed to the [update function](https://github.com/scipp/plopp/blob/main/src/plopp/graphics/graphicalview.py#L126).
That key is either the name of an entry in a dict/DataGroup, or the `uid` of the `Node` that is providing the data.

How about keeping the `uid` argument, but have a default `None` value. Inside the artist, if it is `None`, we then generate an id from `uuid`?